cdap-common/bin/common.sh

```diff
@@ -272,6 +273,10 @@ cdap_set_hive_classpath() {
         # tez-site.xml also need to be passed to explore service
         EXPLORE_CONF_FILES=${EXPLORE_CONF_FILES}:${TEZ_CONF_DIR}/tez-site.xml:
       fi
+      if [[ "${HIVE_EXEC_ENGINE}" == "spark" ]]; then
```

Comment:
So, we actually do. The rest of the changes do the Spark checks/setup before we setup Hive. This still allows Spark to be optional. Otherwise, it'll make Spark a hard dependency for Explore.